pkg/wallet/transactions/standardtx.go

```diff
@@ -266,6 +266,12 @@ func (s *Standard) prove(hasher func() ([]byte, error), encryptValues bool) erro
 	if err != nil {
 		return err
 	}
+	// Remove this has from the TxID field - the hash will be changed after the proof
+	// is done.
+	// TODO: CalculateHash should not set anything, as it's name implies only the
+	// calculation of the hash. This should be adjusted and the rest of the code
```

Comment:
If CalculateHash is refactored to not-set-txid idea, we'd need to ensure no perf regression occurs as a result. I used to observe perf impact in the past due to not-saved-txid and recalculating it here and there.

Reply:
Good point. I will open an issue for it and include your comment.